### ISSUE TITLE
Conflict between `{styler}` and `{lintr}` about braces inside ggplot2 chained calls

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2853]
cc @lorenzwalthert

IMHO, `{styler}` behaviour is the correct one from readability perspective.

## Code with lint

``` r
"library(ggplot2)
x <- TRUE
ggplot(mtcars, aes(wt, mpg)) +
  {
    if (x) geom_point()
  } +
  geom_smooth()" -> code

library(lintr)
lint(text = code, linters = brace_linter())
#> <text>:4:3: style: [brace_linter] Opening curly braces should never go on their own line and should always be followed by a new line.
#>   {
#>   ^
```

## Code without lint

If I adjust the brace position to pacify this linter, it works:

```r
"library(ggplot2)
x <- TRUE
ggplot(mtcars, aes(wt, mpg)) + {
    if (x) geom_point()
  } +
  geom_smooth()" -> code

library(lintr)
lint(text = code, linters = brace_linter())
#> ℹ No lints found.
```

But this code, when submitted to `{styler}`, produces the same code as above; i.e., code that produces a lint:

```r
styler::style_text(code)
#> library(ggplot2)
#> x <- TRUE
#> ggplot(mtcars, aes(wt, mpg)) +
#>   {
#>     if (x) geom_point()
#>   } +
#>   geom_smooth()
```

<sup>Created on 2025-04-23 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
Related: #1440
